modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectViewLocalServiceImpl.java

```diff
@@ -286,15 +301,67 @@ private List<ObjectViewSortColumn> _addObjectViewSortColumns(
 			});
 	}
 
-	private void _validate(
+	private void _validate(long objectViewId, long objectDefinitionId)
+		throws PortalException {
+
+		ObjectView objectView = objectViewPersistence.fetchByODI_DOV_First(
+			objectDefinitionId, true, null);
+
+		if ((objectView != null) &&
+			(objectView.getObjectViewId() != objectViewId)) {
+
+			throw new DefaultObjectViewException(
+				"There can only be one default object view");
+		}
+	}
+
+	private void _validateObjectViewColumn(
+			ObjectView objectView, List<ObjectViewColumn> objectViewColumns)
+		throws ObjectViewColumnFieldNameException {
+
+		List<ObjectField> objectFields =
+			_objectFieldPersistence.findByObjectDefinitionId(
+				objectView.getObjectDefinitionId());
+
+		Set<String> objectFieldNames = new LinkedHashSet<>();
+
+		objectFieldNames.addAll(
+			SetUtil.fromArray(
+				"creator", "dateCreated", "dateModified", "id", "status"));
+
+		objectFields.forEach(
+			objectField -> objectFieldNames.add(objectField.getName()));
+
+		Set<String> objectViewColumnFieldNames = new LinkedHashSet<>();
+
+		for (ObjectViewColumn objectViewColumn : objectViewColumns) {
+			if (!objectFieldNames.contains(
+					objectViewColumn.getObjectFieldName())) {
+
+				throw new ObjectViewColumnFieldNameException(
+					"There is no object field with the name: " +
+						objectViewColumn.getObjectFieldName());
+			}
+
+			if (objectViewColumnFieldNames.contains(
+					objectViewColumn.getObjectFieldName())) {
+
+				throw new ObjectViewColumnFieldNameException(
+					"Already exist an object view column with field name: " +
+						objectViewColumn.getObjectFieldName());
+			}
+
+			objectViewColumnFieldNames.add(
+				objectViewColumn.getObjectFieldName());
+		}
+	}
+
+	private void _validateObjectViewSortColumn(
```

Comment:
Since the parameter is a list (List), could you rename the method to the plural (_validateObjectViewSortColumns) ?

Reply:
Ok